src/coreclr/pal/src/debug/debug.cpp

```diff
@@ -742,15 +742,19 @@ PAL_ProbeMemory(
     BOOL fWriteAccess)
 {
     int fds[2];
+    int flags;
 
     if (pipe(fds) != 0)
     {
         ASSERT("pipe failed: errno is %d (%s)\n", errno, strerror(errno));
         return FALSE;
     }
 
-    fcntl(fds[0], O_NONBLOCK);
-    fcntl(fds[1], O_NONBLOCK);
+    flags = fcntl(fds[0], F_GETFL, 0);
+    fcntl(fds[0], F_SETFL, flags | O_NONBLOCK);
```

Comment:
an alternative would be using `pipe2`  but this would require an `#ifdef `

Reply:
Do some targets not support `pipe2`?

Reply:
`pipe2` is Linux-specific and we support macOS and few other Unix-like OSese. The current solution is good, I am going to merge the PR.